<?xml version="1.0"?>
<launch>
    <arg name="ns"            default="stereo_pair"/>
    <arg name="tf_prefix"     default="charuco"/>
    <arg name="image_left"    default="left/image_rect_color"/>
    <arg name="image_right"   default="right/image_rect_color"/>
    <arg name="markers"       default="charuco"/>
    <arg name="stereo_info"   default="stereo_info"/>

<!--     <arg name="ns"            default="camera"/> -->
<!--     <arg name="tf_prefix"     default="charuco"/> -->
<!--     <arg name="image"         default="color/image_raw"/> -->
<!--     <arg name="markers"       default="charuco"/> -->
<!--     <arg name="camera_info"   default="/camera/color/camera_info"/> -->


    <arg name="display"       default="true"/>
    <arg name="dictionary"    default="3"/>

    <arg name="board_width"    default="5"/>
    <arg name="board_height"   default="4"/>
    <arg name="square_length" value="70.0"/><!--mm-->
    <arg name="marker_length" value="54.0"/><!--mm-->

  <arg name="is_depth_in_meters" default="true" />

  <group ns="$(arg ns)">
      <node name="charcuo_detector_node" pkg="aruco_detector" type="stereo_detector_node" output="screen">
        <param name="type" value="charuco"/>

        <param name="image_left"    value="$(arg image_left)" />
        <param name="image_right"   value="$(arg image_right)" />
        <param name="markers"       value="$(arg markers)"/>
        <param name="stereo_info"   value="$(arg stereo_info)" />
        <param name="tf_prefix"     value="$(arg tf_prefix)" />
        <param name="display"       value="$(arg display)"/>
        <param name="dictionary"    value="$(arg dictionary)"/>

        <param name="board_width"    value="$(arg board_width)"/>
        <param name="board_height"   value="$(arg board_height)"/>
        <param name="marker_length"  value="$(arg marker_length)"/>
        <param name="square_length"  value="$(arg square_length)"/>

        <param name="is_depth_in_meters" type="bool" value="$(arg is_depth_in_meters)" />
      </node>
   </group>
</launch>